src/app/providers/orientation-monitor/orientation-monitor.provider.ts

```diff
@@ -0,0 +1,40 @@
+import { ApplicationRef, Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { ScreenOrientation } from '@capawesome/capacitor-screen-orientation';
+import { isPortrait } from '@shared/helpers/is-portrait-mode';
+
+@Injectable()
+export class OrientationMonitorProvider {
+
+  isPortraitMode$: BehaviorSubject<boolean> = new BehaviorSubject(false);
+
+  constructor(
+    private ref: ApplicationRef,
+  ) {
+  }
+
+  async ionViewWillLeave() {
```

Comment:
Just checking this actually fires? Wasnt aware life cycle hooks applied to providers